<launch>
  /* <param name="robot_description" command="cat $(find nextcode_braccio)/urdf/braccio_arm.urdf"/> */

  <!-- Launch the Move it pipeline and connect to the robot arm using rosserial -->
  /* <arg name="capability_group" default="braccio_arm"/> */
  /* <arg name="use_joint_state_gui" default="false"/> */
  /* <arg name="use_usb_camera" default="true"/> */
  <arg name="cam_device" default="/dev/video0"/>

  /* <!-- Broadcast static tf for robot root --> */
  /* <node pkg="tf" type="static_transform_publisher" name="tf_odom_base_link" args="0 0 0 0 0 0 odom base_link 100"> */
  /*   <param name="capability_group" value="$(arg capability_group)"/> */
  /* </node> */

  /* <!-- Publish tf for the robot links --> */
  /* <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen" clear_params="true"> */
  /*   <param name="capability_group" value="$(arg capability_group)"/> */
  /*   <remap from="joint_states" to="current_state" /> */
  /* </node> */

  <!-- Start ROS communication between the arm and PC -->
  <node pkg="rosserial_python" name="rosserial_braccio" type="serial_node.py" output="screen" clear_params="true">
    <param name="capability_group" value="$(arg capability_group)"/>
    <param name="port" value="/dev/ttyACM0" />
    <param name="baud" value="115200" />
  </node>

  <!-- Joint interface -->
  <node name="DirectJointInterface" pkg="nextcode_braccio" type="braccio_interface.py" output="screen" clear_params="true">
    <param name="camera_width" value="1280" />
    <param name="camera_height" value="720" />
    <param name="camera_radius" value="5" />
    <param name="capability_group" value="$(arg capability_group)"/>
    <param name="kill_on_stop" value="100" />
    <param name="goal_tolerance" value="0.05" />
    <param name="p_gain_yaw" value="0.0005" />
    <param name="p_gain_pitch" value="0.0005" />
    /* <rosparam param="initial_pose">[2.0, 2.2, 0.7, 0.5, 1.7, 1.27]</rosparam> */
    <rosparam param="initial_pose">[2.8, 2.2, 1.0, 1.0, 1.7, 1.27]</rosparam>
  </node>

</launch>
